### DISCUSSION TITLE
Odd image paths in the index.xml file.

### DISCUSSION THREAD

**Discussion by graphdr** [quarto-dev/quarto-cli#946] [Q&A]
I'm porting my blog from distill to quarto and publishing it using GitHub pages. Very happy with the process overall! 

The odd thing is that the index.xml for the RSS feed has image paths with backslashes instead of forward slashes that cause the images to be broken when viewed via an RSS checker. An example of a path from my XML file is: 

<img src="https://github_pages_url/posts\name_of_post\index_files\figure-html\name_of_image.png" class="img-fluid" width="672">

When I manually edit the backslashes in the index.xml file, converting them to forward slashes, the image correctly appears in the SimplePie feed checker. Otherwise, the images are broken. 

I compared my _quarto.yml, _metadata.yml, and index.qmd files to those of Bryan Hanson's blog (also using quarto via GitHub) and couldn't find any obvious difference, yet when I run his XML file through SimplePie, his images are fine. 

Any guidance you can provide would be most appreciated!

**Comment by graphdr** [comment]
Sorry about that. I should have formatted the HTML example image path. Here it is:

`<img src="https://graphdr.github.io/data-stories/posts\2022-02-19-facets-not-small-multiples\index_files\figure-html\chart-assignment-1-1.png" class="img-fluid" width="672">`

**Reply by bryanhanson** [reply]
Interesting that the slashes are forward until you reach the post dir and then back slashes.

**Comment by bryanhanson** [comment]
FWIW, my blog is built on a Mac (macOS 12.4).  My stuff is at https://github.com/bryanhanson/CSR if anyone needs to inspect.  Could this be a Windows issue?

**Comment by dragonstyle** [comment]
Welp that isn’t right! I’ll take a look!

**Comment by dragonstyle** [comment]
I just cloned your repo and took a quick look. Everything looked good to me when I rendered with the latest version of Quarto. I wonder if perhaps it a bug in an earlier version of Quarto which has been fixed. Would you mind confirming the version you're on and if necessary, trying the latest version? (I did make a small fix to address duplicate slashes in paths inside of feeds, but that shouldn't have deal with incorrect slashes, just duplicates).

(I actually just found your blog online - looks like you're building with the dev version of Quarto (cool!) so just pulling the latest from main should ideally get you working if my theory is correct).

Let me know what you find and if that doesn't resolve it I'll keep digging!

**Reply by graphdr** [reply]
I downloaded the main version (quarto-0.9.440-win), rebooted my computer, rendered the site, pushed to GitHub, problem persists.

**Comment by graphdr** [comment]
I found another instance of the same issue. If you scroll down to the bottom of a post, the Citation block has the same problem.

**Reply by dragonstyle** [reply]
Ok let me try again. Any chance you can share a sample qmd that exhibits the problem? Citation block I’ll likely show me the issue if you have a simple page with that…

**Reply by graphdr** [reply]
Sure, Ill give it a go.

**Comment by graphdr** [comment]
Following the [create a blog instructions](https://quarto.org/docs/websites/website-blog.html#blog-projects) I created a new repo [quarto-forwardslash-reprex](https://github.com/graphdr/quarto-forwardslash-reprex). 

I made as few changes as possible to the default blog structure: 
- deleted one of the default posts
- edited the default `post-with-code` post with a short chunk of R code to create a plot

- In the main `index.qmd` I added the lines
```
categories: true
  feed:
    categories: R
```

- In the post `index.qmd` file, I edited the categories line
```
title: "Post With Code"
author: "Harlow Malloc"
date: "2022-05-22"
categories: [R]
image: "image.jpg"
```

 - to `_metadata.yml` I added the line
```
citation: true
```

- edited `_quarto.yml` to change the output to docs, edited the title and description, added a site URL, and an RSS icon in the navbar
```
project:
  type: website
  output-dir: docs

website:
  title: "quarto-forwardlash-reprex"
  site-url: https://graphdr.github.io/quarto-forwardslash-reprex
  description: |
    A reprex for the forwardslash in image links 
    in the XML file.

  navbar:
    right:
      - icon: rss
        href: index.xml
```

Rendering the site then looking at the XML file, the forward slashes persist in the image link and the citation link.

```
<img src="https://graphdr.github.io/quarto-forwardslash-reprex/posts\post-with-code\index_files\figure-html\Nile-data-1.png" class="img-fluid" width="672">

<a href="https://graphdr.github.io/quarto-forwardslash-reprex/posts\post-with-code">
```

I hope this will help track down the problem.

**Comment by dragonstyle** [comment]
Ok, I found the issue in the way we're forming absolute urls (joining the site url to path to the image) and it should be fixed as of v0.9 build 441. Thank you for helping me reproduce the issue so I could track it down!

**Comment by graphdr** [comment]
Fantastic!! Thanks.

**Comment by graphdr** [comment]
We've definitely made progress. The image URLs all have backslashes now as expected and the index.xml file now passes muster on the SimplePie RSS feed check. Great! 

However, the link in the citation at the bottom of the post  has forward slashes. Here's the example citation from [https://github.com/graphdr/quarto-forwardslash-reprex](https://github.com/graphdr/quarto-forwardslash-reprex)

```
<div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div>
<div id="ref-malloc2022" class="csl-entry quarto-appendix-citeas">
Harlow Malloc. 2022. 
<span>“Post With Code.”</span> 
May 22, 2022. 
<a href="https://graphdr.github.io/quarto-forwardslash-reprex/posts\post-with-code">
https://graphdr.github.io/quarto-forwardslash-reprex/posts\post-with-code</a>.
</div>
```

**Reply by dragonstyle** [reply]
I made an issue to track this (as it is clearly a bug). I'll take a look and fix it!

#960

**Reply by graphdr** [reply]
Thanks!

**Comment by graphdr** [comment]
Sure, will try that tomorrow.
